### PR TITLE
QT to AV foundation!

### PR DESCRIPTION
**Pull request by shareefalis**
Still need work to be done.

done: fix the format picker for the camera layer.
todo: rename the file
Todo: format the file to belledonne  style

### CODE REVIEW

File: src/videofilters/qtcapture.m [hunk 1]
```diff
@@ -135,17 +138,28 @@ - (void)captureOutput:(AVCaptureOutput *)captureOutput didOutputVideoFrame:(CVIm
...
 		mblk_t *buf=allocb(size,0);
 		memcpy(buf->b_wptr, data, size);
 		buf->b_wptr+=size;
 		putq(&rq, buf);
-	}*/
+        CVPixelBufferLockBaseAddress(frame, 0);
+        GLubyte *rawImageBytes = CVPixelBufferGetBaseAddress(frame);
+         size_t bytesPerRow = CVPixelBufferGetBytesPerRow(frame);
+         NSData *dataForRawBytes = [NSData dataWithBytes:rawImageBytes length:bytesPerRow * CVPixelBufferGetHeight(frame)];
+        uint8_t *dataWithBytes = (uint8_t*)[dataForRawBytes bytes];
```
Comment (shareefalis) [inline src/videofilters/qtcapture.m]: Todo: dataForRawBytes to dataForRawBytesNoCopy

**Comment by kennethdick** [comment]
@Viish @ghismary Is change to AV foundation to support OSX store possible to merge/support?

**Comment by shareefalis** [comment]
@jehandelalune  please see this for alternative for QTCapture. This replaces qt capture with AVfoundation capture, you are free to use my base as a good starting point.

**Comment by jehandelalune** [comment]
Thanks, 
Good start indeed. May I ask you to provide us with a contributor agreement for your patch ?
http://www.belledonne-communications.com/downloads/Belledonne_communications_CA.pdf
Jehan

**Comment by shareefalis** [comment]
Sorry about the long wait.. Will get on it..  where do i send it to?

**Comment by jehandelalune** [comment]
you can send it to me at jehan.monnier [at] belledonne-communications.com

**Comment by BCopeland28** [comment]
@jehandelalune    Jehan I just sent you the contributor agreement. Let me know if any questions.

**Comment by jehandelalune** [comment]
Thanks, I just started the integration based on your patch.

**Comment by ghismary** [comment]
Patch has been integrated.

**Pull request closed without merging**